ironfish-cli/src/commands/chain/genesisblock.ts

```diff
@@ -64,34 +74,119 @@ export default class GenesisBlockCommand extends IronfishCommand {
       this.exit(0)
     }
 
-    let account = null
-    if (flags.account !== null) {
-      account = node.wallet.getAccountByName(flags.account)
-    }
+    const expectedSupply = CurrencyUtils.decodeIron(flags.genesisSupplyInIron)
+    let allocations: GenesisBlockAllocation[]
+    if (flags.allocations) {
+      // If the allocations flag is set, read allocations from a CSV file
+      const csv = await fs.readFile(flags.allocations, 'utf-8')
+      const result = parseAllocationsFile(csv)
+
+      if (!result.ok) {
+        this.error(result.error)
+      }
+
+      const totalSupply: bigint = result.allocations.reduce((prev, cur) => {
+        return prev + cur.amountInOre
+      }, 0n)
+
+      if (totalSupply !== expectedSupply) {
+        this.error(
+          `Allocations file contains ${CurrencyUtils.encodeIron(
+            totalSupply,
+          )} $IRON, but --genesisSupplyInIron expects ${flags.genesisSupplyInIron} $IRON.`,
+        )
+      }
+
+      allocations = result.allocations
```

Comment:
Do you think we need a way to dry run the command with the csv file? So in the dry run, it will print out the parsed csv file

Reply:
Yep, agreed. At least should probably print out addresses and confirm first

Reply:
Good idea, I added a `--dryRun` flag that logs the difficulty and list of allocations, then exits, and I also added a confirmation step to creating the genesis block that logs the difficulty and list of allocations and asks to confirm.

I also added duplicate detection -- It could be intentional that someone would include the same address multiple times, but worth informing the user in case it was a copy/paste error.